doc/rst/language/spec/iterators.rst

```diff
@@ -84,6 +84,13 @@ execution continues from the point immediately following the ``yield``
 statement. A yield statement in an iterator that yields references must
 yield an lvalue expression.
 
+The iterator's ``yield-intent`` determines how each value is yielded.
+The rules for yielding are the same as the rules for returning values
+from procedures, see :ref:`Return_Intent`, except an iterator with
+the ``ref`` or ``const ref`` yield intent is allowed to yield an lvalue
```

Comment:
Is this a behavior that we've agreed we want to continue supporting?

Reply:
We have not had the discussion about changing it. So keeping it this way for now until we decide to change, if at all.